pkg/cloudprovider/cloudprovider.go

```diff
@@ -350,15 +350,15 @@ func (c *CloudProvider) instanceToNodeClaim(ctx context.Context, vm *armcompute.
 	}
 	annotations[v1alpha2.AnnotationInPlaceUpdateHash] = inPlaceUpdateHash
 
-	nodeClaim.Name = GenerateNodeClaimName(*vm.Name)
+	nodeClaim.Name = GenerateNodeClaimName(lo.FromPtr(vm.Name))
 	nodeClaim.Labels = labels
 	nodeClaim.Annotations = annotations
-	nodeClaim.CreationTimestamp = metav1.Time{Time: *vm.Properties.TimeCreated}
+	nodeClaim.CreationTimestamp = metav1.Time{Time: lo.FromPtr(vm.Properties.TimeCreated)}
```

Comment:
this path was executed when my List query returned an empty result. 

In the case where ARG has an outage, garbage collection will break, but there is no need to panic here in that scenario. So switched to lo.FromPtr() instead of normal dereference.

Reply:
just going to revert for now.